pyvista/plotting/axes_assembly.py

```diff
@@ -465,43 +478,212 @@ def label_color(self, color: ColorLike):  # numpydoc ignore=GL08
+        _AxesPropTuple(x_shaft=True, y_shaft=False, z_shaft=True, x_tip=False, y_tip=False, z_tip=False)
+
+        Set :attr:`~pyvista.Property.style` for a single axis and specific part.
+
+        >>> axes_actor.set_part_prop(
+        ...     'style', 'wireframe', axis='x', part='shaft'
+        ... )
+        >>> axes_actor.get_part_prop('style')
+        _AxesPropTuple(x_shaft='Wireframe', y_shaft='Surface', z_shaft='Surface', x_tip='Surface', y_tip='Surface', z_tip='Surface')
+        """
+        actors = self._filter_part_actors(axis=axis, part=part)
+        values: Sequence[float | str | ColorLike]
+
+        # Validate input as a sequence of values
+        if 'color' in name:
+            # Special case for color inputs
+            if axis == 'all' and part == 'all':
+                n_values = 6
+            elif part == 'all':
+                n_values = 2
+            elif axis == 'all':
+                n_values = 3
+            else:
+                n_values = 1
+            values = _validate_color_sequence(value, n_values)
+        elif isinstance(value, Sequence) and not isinstance(value, str):
+            # Number sequence
+            values = value
+        else:
+            # Scalar number or string
+            values = [value] * len(actors)
+
+        if len(values) != len(actors):
+            raise ValueError(
+                f"Number of values ({len(values)}) in {value} must match the number of parts ({len(actors)}) for axis '{axis}' and part '{part}'"
+            )
+
+        # Sequence is valid, now set values
+        for actor, val in zip(actors, values):
+            setattr(actor.prop, name, val)
+
+    def get_part_prop(self, name: str):
+        """Get :class:`~pyvista.Property` attributes for the axes shafts and/or tips.
+
+        This is a generalized getter method which returns the value of
+        a specific :class:`pyvista.Property` attribute for all shafts and tips.
+
+        Parameters
+        ----------
+        name : str
+            Name of the :class:`~pyvista.Property` attribute to get.
+
+        Returns
+        -------
+        tuple
+            Named tuple with attribute values for the axes shafts and tips.
+            The values are ordered ``(x_shaft, y_shaft, z_shaft, x_tip, y_tip, z_tip)``.
+
+        Examples
+        --------
+        Get the ambient property of the axes shafts and tips.
+
+        >>> import pyvista as pv
+        >>> axes_assembly = pv.AxesAssembly()
+        >>> axes_assembly.get_part_prop('ambient')
+        _AxesPropTuple(x_shaft=0.0, y_shaft=0.0, z_shaft=0.0, x_tip=0.0, y_tip=0.0, z_tip=0.0)
+
+        """
+        prop_values = [getattr(actor.prop, name) for actor in self._shaft_and_tip_actors]
+        return _AxesPropTuple(*prop_values)
+
+    def _filter_part_actors(
+        self,
+        axis: Literal['x', 'y', 'z', 0, 1, 2, 'all'] = 'all',
+        part: Literal['shaft', 'tip', 0, 1, 'all'] = 'all',
```

Comment:
I'm thinking we should remove the use of integers from the type annotations and the docstring. They are used interally, so we can support it unofficially, but otherwise they are like magic numbers. And I don't think we should expose the enums either.